### ISSUE TITLE
async

### ISSUE THREAD

**Issue by timeyyy** [kivy/kivy#4799]
I wanted to use an async approach with kivy but found no built in solution. @tshirtman did some great work getting some prototypes. I had a play myself but it was rather involved and i gave up.

Recently my google fu  found an awesome [async gui library](https://github.com/reclosedev/async_gui/pull/4) that i have added kivy support to. 

The library provides an easy way to spin off functions to be run in other threads or even other processes. It also has [documentation](http://async-gui.readthedocs.io/en/latest/) and [tests](https://github.com/reclosedev/async_gui/tree/master/tests). The [code](https://github.com/reclosedev/async_gui/blob/master/async_gui/engine.py) is also well written with doc strings.

It is lacking support for running python coroutines from within these functions though.

TLDR 
This is the most complete solution i have found to using async programming with kivy. This should be documented.

It seems there was [interest](https://bugs.python.org/issue27546) on the python issue tracker about async programming with tkinter. Efforts could be made so a complete async solution exists for all majour gui toolkits on python.

**Comment by dessant** [comment]
@timeyyy nice work! This could be presented on a wiki page, it would be a bit early to add to the docs.
https://github.com/kivy/kivy/wiki

**Comment by timeyyy** [comment]
Oh i didn't even know the wiki existed. Alot of good info there. I added a link to https://github.com/kivy/kivy/wiki/Snippets

**Comment by IgorCode** [comment]
For people who need this for Android, it does not seem to work on Android because it uses multiprocess library.

**Comment by fitoprincipe** [comment]
Hi @timeyyy,
I'm trying to run it, but I have an issue. First, `async_gui` package available on pypi doesn't have the `kivy`module. So, I had to download the source and install it from source. Now, it says

```
  File "/usr/local/lib/python2.7/dist-packages/async_gui/toolkits/kivy.py", line 3, in <module>
    from kivy.base import EventLoop

ImportError: No module named base
```

**Comment by tshirtman** [comment]
Just want to point out there is another, very promising PR (better than my experiments imho), by matham, here, https://github.com/kivy/kivy/pull/5241 that you might want to test.

**Comment by fitoprincipe** [comment]
thank you @tshirtman!

**Comment by timeyyy** [comment]
@fitoprincipe if you are using python2 you will need this https://github.com/reclosedev/async_gui/pull/7
